java/sage/EPG.java

```diff
@@ -1631,25 +1662,25 @@ public long getCachedProviderIDForName(String s)
   private EPGDataSource currDS;
   private boolean alive = true;
   private java.io.File logoDir;
-  private java.util.Map logoMap;
+  private java.util.Map<String, java.io.File> logoMap;
   //	private java.util.Map colorMap;
 
   private Object stationIDCacheLock = new Object();
   private int[] viewableStationIDCache;
 
-  private java.util.Map lineups;
-  private java.util.Map lineupOverrides;
-  private java.util.Map serviceLevels;
+  private final java.util.Map<Long, java.util.Map<Integer, String[]>> lineups;
+  private final java.util.Map<Long, java.util.Map<Integer, String[]>> lineupOverrides;
+  private final java.util.Map<Long, java.util.Map<Long, Integer>> serviceLevels;
```

Comment:
You changed this from being an Integer, Integer map to being a Long, Integer map. It should be an Integer, Integer map because it's mapping station IDs to service levels and station IDs are Integer values....there's a few spots where this needs to be changed to resolve that.